src/motion-queue.h

```diff
@@ -105,9 +105,11 @@ class MotionQueue {
   // Shutdown. If !flush_queue: immediate, even if motors are still moving.
   virtual void Shutdown(bool flush_queue) = 0;
 
-  // Fill the argument with the current absolute position in loops
-  // for each motor.
-  virtual void GetMotorsLoops(MotorsRegister *absolute_pos_loops) = 0;
+  // Return the number of motion segments enqueued after
+  // the one currently executing and its already executed loops.
+  // buffer_size will never be 0 except if nothing has been yet enqueued.
+  virtual void GetExecutionProgress(uint32_t *loop_progress_working_segment,
```

Comment:
In the offline discussion, we had already had some better concept. I think something like
```
   virtual int GetPendingElements(uint32_t *head_item_progress);
```
or something (naming needs improvement). Essentially, since this is expected to behave as a queue, we should have functions that reflect that. This naturally would reflect how much there is to do; if the number returned is non-zero, the queue is working on something and the head_item_progress would tell us how much.